docs/topics/http/urls.txt

```diff
@@ -368,6 +368,21 @@ the following example is valid::
 In the above example, the captured ``"username"`` variable is passed to the
 included URLconf, as expected.
 
+Nested captured parameters, for example ``(?P<outer>(?P<inner>\w+))`` or
+``((\d+)/)?``, are discarded by Django when urls are reversed, using only the
+outer parameters to find a reverse for a pattern. If you want to reverse such
+patterns, only pass the outer argument or named argument.
+
+For cases such as making a part of the pattern optional, use non-capturing
+arguments -- for example when a non-query page-specifier is optional::
+
+    # In urls.py
+    from django.conf.urls import url
+
+    urlpatterns = [
+        url(r'^(?P<username>\w+)/(?:page-(?P<page>\d+)/)?$', blog_entry),
```

Comment:
Could you explain this example a bit more? Maybe showing what the URLs are meant to look like and how to reverse them?

Here are some other suggested edits:
```diff
diff --git a/docs/topics/http/urls.txt b/docs/topics/http/urls.txt
index ab90b90..1703595 100644
--- a/docs/topics/http/urls.txt
+++ b/docs/topics/http/urls.txt
@@ -369,14 +369,13 @@ In the above example, the captured ``"username"`` variable is passed to the
 included URLconf, as expected.
 
 Nested captured parameters, for example ``(?P<outer>(?P<inner>\w+))`` or
-``((\d+)/)?``, are discarded by Django when urls are reversed, using only the
-outer parameters to find a reverse for a pattern. If you want to reverse such
-patterns, only pass the outer argument or named argument.
+``((\d+)/)?``, are discarded by Django when URLs are reversed. Only the
+outer parameters are used to find a match for a pattern. If you want to reverse
+patterns with nested groups, only pass the outer or named argument.
 
 For cases such as making a part of the pattern optional, use non-capturing
-arguments -- for example when a non-query page-specifier is optional::
+arguments -- for example, when a non-query page specifier is optional::
 
-    # In urls.py
     from django.conf.urls import url
 
     urlpatterns = [
```